### PR TITLE
yumpkg: report stdout when stderr is redirected to stdout

### LINKED ISSUE #57862: [BUG] pkg.install on Fedora 32 does not properly report errors
**Description**
`errors` section in pkg.install is empty when an error occurs. The actual error is logged in minion (and dnf), depending on log level.

**Setup**
I've stumbled on it with salt-ssh, but the easiest reproducer is `salt-call --local`.

**Steps to Reproduce the behavior**
```
$ sudo salt-call --local pkg.install non-existing-pkg 
[ERROR   ] Command '['systemd-run', '--scope', 'dnf', '-y', '--best', '--allowerasing', 'install', 'non-existing-pkg']' failed with return code: 1
[ERROR   ] stdout: Running scope as unit: run-rf3cfee1f895647f58b91639ded09691d.scope
Last metadata expiration check: 1:21:11 ago on Fri Jul  3 18:27:01 2020.
No match for argument: non-existing-pkg
Error: Unable to find a match: non-existing-pkg
[ERROR   ] retcode: 1
Error running 'pkg.install': Error occurred installing package(s). Additional info follows:

changes:
    ----------
errors:
```

**Expected behavior**
Error message included in `errors` field (as the message suggests). In this case it would be:
```
No match for argument: non-existing-pkg
Error: Unable to find a match: non-existing-pkg
```

**Versions Report**

Default salt package in Fedora 32.

<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
           Salt: 3001rc1
 
Dependency Versions:
           cffi: 1.14.0
       cherrypy: Not Installed
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.11.2
        libgit2: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.2
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: Not Installed
   pycryptodome: 3.9.7
         pygit2: Not Installed
         Python: 3.8.3 (default, May 29 2020, 00:00:00)
   python-gnupg: Not Installed
         PyYAML: 5.3.1
          PyZMQ: 18.1.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2
 
System Versions:
           dist: fedora 32 
         locale: utf-8
        machine: x86_64
        release: 5.4.41-1.qubes.x86_64
         system: Linux
        version: Fedora 32 

```
</details>

### PR DESCRIPTION
**Pull request by marmarek**
### What does this PR do?

When 'redirect_stderr=True' is set, stderr is empty and both
stdout+stderr are logged into stdout. Include the right one in the
output.

### What issues does this PR fix or reference?
Fixes: 57862

### Previous Behavior
Empty `errors` field on errors.

### New Behavior
`errors` field populated.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by cmcmarrow** [comment]
@marmarek I know it’s a little redundant but can we get a test to verify that does not turn back into a regression again? Also thanks for the PR! The failing test are not related to you at this time. It’s a problem on are end.

**Review by cmcmarrow — CHANGES_REQUESTED** [review]
I think it would be good to include `stdout` and `stderr`. I think someone changed it to `stdout` because they were not getting info need from `stderr`

**Comment by marmarek** [comment]
> I think it would be good to include `stdout` and `stderr`

In this specific case it doesn't make sense, as stderr is redirected to stdout. Maybe two other cases should also have `redirect_stderr=True` instead of logging stdout and stderr separately?

**Comment by terminalmage** [comment]
@marmarek Yeah, I think I mistakenly "corrected" the line you're fixing in this PR, I should have left it alone. Can you revert the other two from b40e2128 but also add `redirect_stderr=True` to the `_call_yum()` function call? That seems like the best fix here.

**Comment by terminalmage** [comment]
@marmarek Just checking back in since I haven't heard back from you. Can you make those changes I mentioned in my last comment?

**Comment by marmarek** [comment]
I've finally found some time to finish this, sorry for the delay.

**Comment by terminalmage** [comment]
Thanks! This looks good, just need a few fixes identified by pre-commit.

**Comment by marmarek** [comment]
pre-commit seems to look at the whole file, not only changed parts. Should I include reformatting and various other updates like removing `six` in the commit? Sounds unrelated to this PR...

**Comment by terminalmage** [comment]
I know that the core team will not merge this without the tests failing. It seems that the failures are related to some automated code reformatting to ditch usage of six and associated code conventions that ensure Python 2.7 compatibility, now that Python 2 support has been dropped.

Perhaps one of the core team can apply the pre-commit hooks and push to this PR branch.

**Review by cmcmarrow — DISMISSED** [review]

**Comment by Ch3LL** [comment]
I went ahead and ran pre-commit and fixed some merge conflicts on rebase.

**Comment by Ch3LL** [comment]
ping @terminalmage would you mind re-reviewing since I have technically pushed code to this now would be nice to get one more review.

**Review by Ch3LL — DISMISSED** [review]

**Review by terminalmage — DISMISSED** [review]
Looks great :+1:

**Pull request merged**
